crates/astria-core/src/crypto.rs

```diff
@@ -0,0 +1,83 @@
+use std::fmt::{
+    self,
+    Debug,
+    Formatter,
+};
+
+use ed25519_consensus::{
+    Signature,
+    SigningKey as Ed25519SigningKey,
+    VerificationKey,
+    // VerificationKeyBytes,
+};
+use rand::{
+    CryptoRng,
+    RngCore,
+};
+use zeroize::{
+    Zeroize,
+    ZeroizeOnDrop,
+};
+
+#[derive(Clone, Zeroize, ZeroizeOnDrop)]
+pub struct SigningKey(Ed25519SigningKey);
+
+impl SigningKey {
+    /// Generates a new signing key.
+    pub fn new<R: RngCore + CryptoRng>(rng: R) -> Self {
+        Self(Ed25519SigningKey::new(rng))
+    }
+
+    /// Creates a signature on `msg` using this key.
+    #[must_use]
+    pub fn sign(&self, msg: &[u8]) -> Signature {
+        self.0.sign(msg)
+    }
+
+    /// Returns the byte encoding of the signing key.
+    #[must_use]
+    pub fn to_bytes(&self) -> [u8; 32] {
+        self.0.to_bytes()
+    }
+
+    /// Returns the byte encoding of the signing key.
+    #[must_use]
+    pub fn as_bytes(&self) -> &[u8; 32] {
+        self.0.as_bytes()
+    }
+
+    /// Returns the verification key associated with this signing key.
+    #[must_use]
+    pub fn verification_key(&self) -> VerificationKey {
+        self.0.verification_key()
+    }
+}
+
+impl Debug for SigningKey {
```

Comment:
Yeah ok - so we hide the signing key, but still allow for indirectly checking it by exposing the verification key. That's smart.